src/sentry/api/endpoints/organization_group_index_stats.py

```diff
@@ -0,0 +1,127 @@
+from sentry.api.helpers.group_index import (
+    build_query_params_from_request,
+    rate_limit_endpoint,
+)
+from sentry.api.serializers import serialize
+from sentry.api.serializers.models.group import StreamGroupSerializerSnuba
+from sentry.api.utils import get_date_range_from_params, InvalidParams
+from sentry.models import Group
+from sentry.utils.compat import map
+from sentry.api.endpoints.organization_group_index import ERR_INVALID_STATS_PERIOD
+
+
+class OrganizationGroupIndexStatsEndpoint(OrganizationEventsEndpointBase):
+    permission_classes = (OrganizationEventPermission,)
+
+    @rate_limit_endpoint(limit=10, window=1)
+    def get(self, request, organization):
+        """
+        Get the stats on an Organization's Issues
+        `````````````````````````````
+        Return a list of issues (groups) with the requested stats.  All parameters are
+        supplied as query string parameters.
+
+        :qparam list groups A list of group ids
+        :qparam list expand an optional list of strings to opt in to additional data. Supports `inbox`
+        :qparam list collapse an optional list of strings to opt out of certain pieces of data. Supports `stats`, `lifetime`, `filtered`, and `base`
+
+        The ``groupStatsPeriod`` parameter can be used to select the timeline
+        stats which should be present. Possible values are: '' (disable),
+        '24h', '14d'
+
+        The ``statsPeriod`` parameter can be used to select a date window starting
+        from now. Ex. ``14d``.
+
+        The ``start`` and ``end`` parameters can be used to select an absolute
+        date period to fetch issues from.
+
+        :qparam string statsPeriod: an optional stat period (can be one of
+                                    ``"24h"``, ``"14d"``, and ``""``).
+        :qparam string groupStatsPeriod: an optional stat period (can be one of
+                                    ``"24h"``, ``"14d"``, and ``""``).
+        :qparam string start:       Beginning date. You must also provide ``end``.
+        :qparam string end:         End date. You must also provide ``start``.
+        """
+
+        stats_period = request.GET.get("groupStatsPeriod")
+        try:
+            start, end = get_date_range_from_params(request.GET)
+        except InvalidParams as e:
+            raise ParseError(detail=six.text_type(e))
+
+        expand = request.GET.getlist("expand", [])
+        collapse = request.GET.getlist("collapse", ["base"])
+        has_inbox = features.has("organizations:inbox", organization, actor=request.user)
+        projects = self.get_projects(request, organization)
+        project_ids = [p.id for p in projects]
+
+        try:
+            group_ids = set(map(int, request.GET.getlist("groups")))
+        except ValueError:
+            return Response({"detail": "Group ids must be integers"}, status=400)
+
+        if not group_ids:
+            return Response(
```

Comment:
You could use `rest_framework.exceptions.ParseError` to make these error responses a bit more terse.